witchcraft-env-logger/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+//! ```
+//!
+//! ```not_rust
+//! $ RUST_LOG=info ./main
+//! {"type":"service.1","level":"ERROR","time":"2025-05-26T16:46:31.043928664Z","origin":"main","thread":"main","message":"this is printed by default","safe":true,"params":{"file":"witchcraft-env-logger/examples/main.rs","line":7}}
+//! {"type":"service.1","level":"INFO","time":"2025-05-26T16:46:31.043976765Z","origin":"main","thread":"main","message":"figured out the answer","safe":true,"params":{"answer":12,"file":"witchcraft-env-logger/examples/main.rs","line":11}}
+//! ```
+//!
+//! ```not_rust
+//! $ RUST_LOG=main=debug ./main
+//! {"type":"service.1","level":"DEBUG","time":"2025-05-26T16:47:04.831643644Z","origin":"main","thread":"main","message":"this is a debug message","safe":true,"params":{"file":"witchcraft-env-logger/examples/main.rs","line":6}}
+//! {"type":"service.1","level":"ERROR","time":"2025-05-26T16:47:04.831691314Z","origin":"main","thread":"main","message":"this is printed by default","safe":true,"params":{"file":"witchcraft-env-logger/examples/main.rs","line":7}}
+//! {"type":"service.1","level":"INFO","time":"2025-05-26T16:47:04.831720469Z","origin":"main","thread":"main","message":"figured out the answer","safe":true,"params":{"answer":12,"file":"witchcraft-env-logger/examples/main.rs","line":11}}
+//! ```
+#![warn(missing_docs)]
+
+use std::{
+    env,
+    io::{self, Write},
+};
+
+use conjure_serde::json;
+use witchcraft_log::{LevelFilter, Log, Metadata, Record, SetLoggerError};
+use witchcraft_log_util::{filter::Filter, service};
+
+struct Logger {
+    filter: Filter,
+}
+
+impl Log for Logger {
+    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
+        self.filter.enabled(metadata)
+    }
+
+    fn log(&self, record: &Record<'_>) {
+        if !self.enabled(record.metadata()) {
+            return;
+        }
+
+        let service_log = service::from_record(record);
+        let mut buf = json::to_string(&service_log).unwrap();
+        buf.push('\n');
+        // Using the macro so output is intercepted in tests properly
+        eprint!("{buf}");
+    }
+
+    fn flush(&self) {
+        let _ = io::stderr().flush();
+    }
+}
+
+/// Initializes the global logger, reading configuration from the `RUST_LOG` environment variable.
+///
+/// Returns an error if the logger is already initialized.
+pub fn try_init() -> Result<(), SetLoggerError> {
+    let mut builder = Filter::builder();
+
+    if let Ok(rust_log) = env::var("RUST_LOG") {
+        for directive in rust_log.split(",") {
+            let mut it = directive.splitn(2, "=");
+            let first = it.next().unwrap();
+            let second = it.next();
+
+            match second {
+                Some(level) => {
+                    let Ok(level) = level.parse::<LevelFilter>() else {
+                        continue;
+                    };
+                    builder = builder.target_level(first, level);
```

Comment:
I'm curious why you opted for this structure rather than something like the following:
```suggestion
                    if let Ok(level) = level.parse::<LevelFilter>() {
                        builder = builder.target_level(first, level);
                    };
```